### PR TITLE
Add Payment Frequency Filter

### PR DESCRIPTION
**Pull request by wjpurdum**
# Overview

Describe the change, any relevant motivation and context, and any dependencies it impacts.
Then fill out the checklist below by replacing the [ ] empty checkbox with [x].

## Relevant Issue(s)
Updated the dropdown for frequency of payments filter - so that state is tracked for the input and the "clear all" functionality works. 

Link to the Asana issue this PR addresses
https://app.asana.com/0/0/1200435319008236/f

## Type of change

- [x] Filtering (AirTable to Mapbox)
- [ ] Layering
- [ ] Mapbox Functionality (Must Have)
- [ ] UX/UI
- [ ] Softr Integration
- [ ] Other "Nice to Have" Features

# Self Review Checklist

Perform a self review of your code using the below guidelines.
If you feel your code does not meet the standards, describe why and the reviewer can work with you to address your concerns.

- [ ] My code is DRY ("Do not Repeat Yourself"), i.e. my code is modular and not redundant
- [ ] I have commented my code where useful
- [ ] The variables in my code are named clearly and convey intent
- [ ] There are no obvious optimizations I could make that would improve performance
- [ ] I have removed all logging and debugging from my code
- [ ] I have handled edge cases

### CODE REVIEW

File: src/components/filters/DisbursementFrequency.jsx [hunk 1]
```diff
@@ -1,51 +1,135 @@
...
         return {
             name: strings[key],
             id: key
         }
     })
+
+
+    const handleChange = (e) => {
+        const values = e.target.value;
+        const onFilters = disbursementFrequency && Object.keys(disbursementFrequency).length && Object.keys(disbursementFrequency).map(key => key);
```
Comment (ecaoile) [inline src/components/filters/DisbursementFrequency.jsx:57]: As I discussed in Slack, there's an issue with the values array getting out of sync with the filters object. Changing line 57 to this seems to do the trick:

`const onFilters = disbursementFrequency && Object.keys(disbursementFrequency).length ? Object.keys(disbursementFrequency).map(key => key) : [];`

There are other ways to optimize the handleChange, but this handleChange is a bit cryptic, and I need to make optimization isn't breaking anything.

**Review by ecaoile — CHANGES_REQUESTED** [review]

File: src/components/filters/DisbursementFrequency.jsx (hunk 1 above)
Comment (ecaoile) [reply src/components/filters/DisbursementFrequency.jsx:57]: Update: I lied. Still getting issues when clicking quickly. I'll keep debugging...

**Comment by wjpurdum** [comment]
I can look at it today, might be easier for me to debug since I wrote it.
I'll test out some scenarios.

**Review by ecaoile — APPROVED** [review]

**Comment by ecaoile** [comment]
Going to approve with the note that we still need to figure out a fix for the bug

**Pull request merged**